pkg/datapath: Remove transitive dependency on netlink

aditighag opened the pull request:
See commit message.

ti-mo commented on pkg/k8s/factory_functions.go:
```diff
@@ -8,7 +8,7 @@ import (
 
 	"github.com/cilium/cilium/pkg/comparator"
-	"github.com/cilium/cilium/pkg/datapath"
+	types2 "github.com/cilium/cilium/pkg/datapath/types"
```
Maybe something like `dptypes` instead? Not sure why this import name was chosen. We should also try to use the same import name across all consuming packages.

aditighag replied:
A different import alias was selected because it conflicted with - "github.com/cilium/cilium/pkg/k8s/types"

>We should also try to use the same import name across all consuming packages.

How can you enforce that? I would rather not change it by hand at all the places. I can fix just the one place in `pkg/k8s/factory_functions.go` where there is a conflict. Alternatively, we can rename the `pkg/datapath/types` to `pkg/datapath/dptypes` or something along those lines. But the repetition looks a bit weird.

Edit : I manually renamed the alias in this file to `dpTypes` in the latest push.

sayboras replied:
curious on how we can enforce this. I think right now, it's just manual effort only. However, it's not uncommon in cilium code base.

```
nodeTypes "github.com/cilium/cilium/pkg/node/types"
cnitypes "github.com/containernetworking/cni/pkg/types"
eniTypes "github.com/cilium/cilium/pkg/aws/eni/types"
ipcachetypes "github.com/cilium/cilium/pkg/ipcache/types"
```

aditighag replied:
@ti-mo Ping?

Edit : Review comment has been addressed, moving forward with the merge.

gandro reviewed (approved):
Looks good to me, besides what Timo already pointed out (i.e. a better and consistent import alias such as `dpTypes` or `datapathTypes` would be great)

aditighag committed:
pkg/datapath: Remove transitive dependency on netlink

Ginkgo tests have a transitive dependency on the
netlink package which can lead to failures when
run on macOS. Tobias noticed that the dependency
is coming from pkg/k8s that imports pkg/datapath.
Remove the unnecessary dependency on the netlink
package as we support running the tests on macOS.

Suggested-by: Tobias Klauser <tobias@cilium.io>
Signed-off-by: Aditi Ghag <aditi@cilium.io>
